tools/sctool/src/main/java/com/horizen/CommandProcessor.java

```diff
@@ -598,12 +604,25 @@ private void processGenesisInfo(JsonNode json) {
             int withdrawalEpochLength;
             String sidechainBlockHex;
 
+
             // are we building a utxo or account model based block?
-            if (block_version == AccountBlock.ACCOUNT_BLOCK_VERSION()){
+            if (block_version == AccountBlock.ACCOUNT_BLOCK_VERSION()) {
+
+                byte[] stateRoot;
+                try {
+                    stateRoot = getGenesisStateRoot(mainchainBlockReferencesData, params);
+                }
+                catch (Exception e) {
+                    printer.print(String.format("Error: 'Could not get genesis state root: %s", e.getMessage()));
+                    return;
+                }
 
-                byte[] stateRoot = new byte[MerkleTree.ROOT_HASH_LENGTH];
-                byte[] receiptsRoot = new byte[MerkleTree.ROOT_HASH_LENGTH];
-                AddressProposition forgerAddress = new AddressProposition(new byte[Account.ADDRESS_SIZE]);
+                byte[] receiptsRoot = StateDB.EMPTY_ROOT_HASH; // empty root hash (no receipts)
+
+                AddressProposition forgerAddress =
+                  new AddressProposition(Arrays.copyOf(
+                          sidechainCreation.getScCrOutput().address(),
+                          com.horizen.account.utils.Account.ADDRESS_SIZE)); // taken from the creation
```

Comment:
This is a tricky point. I mean, which part of original 32 bytes take as an eth address.
In the AccountStateView we have the following code:
`val ownerAddressProposition = new AddressProposition(BytesUtils.reverseBytes(scOut.address.take(com.horizen.account.utils.Account.ADDRESS_SIZE)))`
Let's add a utility function that will extract EthAddress from FT and ScCreation address field.
Then use it in all places: here and twice in `applyMainchainBlockReferenceData`

Reply:
Ok, done